### ISSUE TITLE
Docker build - Permission error

### ISSUE THREAD

**Issue by Killuak11** [CTFd/CTFd#2298]
Using docker build, permission problem

- CTFd Version/Commit: 3.5.2
- Operating System: Ubuntu-5.15.0-71-generic
- Web Browser and Version:  Docker version 20.10.21, build 20.10.21-0ubuntu1~22.04.3

**What happened?**
Builds correctly but exits with a permission error when starting containers

**What did you expect to happen?**
Containers to run but keeps failing every time, while docker-compose works fine

**How to reproduce your issue**
git clone #URL####
docker build 
docker build -t testing:latest ./CTFd-master
docker run -d --name test -p 80:8000 testing:latest

** How I Fixed my issue**
I fixed it with one change in the Dockerfile. could be a new error but wanted to give it to others just in case someone else has the same problem.
Dockerfile: line 47 : add parameter : /opt/CTFd

**Any associated stack traces or error logs**
below is the docker log for the container
```
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running stamp_revision  -> 46a278193a94
Traceback (most recent call last):
  File "/opt/CTFd/manage.py", line 19, in <module>
    app = create_app()
  File "/opt/CTFd/CTFd/__init__.py", line 243, in create_app
    cache.init_app(app)
  File "/opt/venv/lib/python3.9/site-packages/flask_caching/__init__.py", line 205, in init_app
    self._set_cache(app, config)
  File "/opt/venv/lib/python3.9/site-packages/flask_caching/__init__.py", line 231, in _set_cache
    app.extensions["cache"][self] = cache_obj(
  File "/opt/venv/lib/python3.9/site-packages/flask_caching/backends/__init__.py", line 68, in filesystem
    return FileSystemCache(*args, **kwargs)
  File "/opt/venv/lib/python3.9/site-packages/flask_caching/backends/filesystem.py", line 74, in __init__
    os.makedirs(self._path)
  File "/usr/local/lib/python3.9/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/usr/local/lib/python3.9/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/opt/CTFd/CTFd/../.data
```

**Comment by carlfugate** [comment]
I am having the same issue.  Was deploying on Docker Desktop and ECS.

2023-05-04 17:59:05 INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
2023-05-04 17:59:05 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
2023-05-04 17:59:05 Traceback (most recent call last):
2023-05-04 17:59:05   File "/opt/CTFd/manage.py", line 19, in <module>
2023-05-04 17:59:05     app = create_app()
2023-05-04 17:59:05   File "/opt/CTFd/CTFd/__init__.py", line 243, in create_app
2023-05-04 17:59:05     cache.init_app(app)
2023-05-04 17:59:05   File "/opt/venv/lib/python3.9/site-packages/flask_caching/__init__.py", line 205, in init_app
2023-05-04 17:59:05     self._set_cache(app, config)
2023-05-04 17:59:05   File "/opt/venv/lib/python3.9/site-packages/flask_caching/__init__.py", line 231, in _set_cache
2023-05-04 17:59:05     app.extensions["cache"][self] = cache_obj(
2023-05-04 17:59:05   File "/opt/venv/lib/python3.9/site-packages/flask_caching/backends/__init__.py", line 68, in filesystem
2023-05-04 17:59:05     return FileSystemCache(*args, **kwargs)
2023-05-04 17:59:05   File "/opt/venv/lib/python3.9/site-packages/flask_caching/backends/filesystem.py", line 74, in __init__
2023-05-04 17:59:05     os.makedirs(self._path)
2023-05-04 17:59:05   File "/usr/local/lib/python3.9/os.py", line 215, in makedirs
2023-05-04 17:59:05     makedirs(head, exist_ok=exist_ok)
2023-05-04 17:59:05   File "/usr/local/lib/python3.9/os.py", line 225, in makedirs
2023-05-04 17:59:05     mkdir(name, mode)
2023-05-04 17:59:05 PermissionError: [Errno 13] Permission denied: '/opt/CTFd/CTFd/../.data'

**Comment by carlfugate** [comment]
> @carlfugate try the fix i did. Dockerfile: line 47 : add parameter : /opt/CTFd

Yes, that worked for me as well.  Thanks @Killuak11

**Comment by carlfugate** [comment]
Submitted PR #2299

**Comment by ColdHeat** [comment]
Thanks for reporting this. What is odd is that this should technically be handled by https://github.com/CTFd/CTFd/blob/master/Dockerfile#L39. And I imagine that if this did break builds for others there would be more reports...

**Comment by ColdHeat** [comment]
I see the difference is that --chown does not set the permissions of the actual /opt/CTFd directory whereas the fix here does and this allows flask_caching to create its cache directory.

**Comment by ColdHeat** [comment]
Closed in #2299

**Comment by carlfugate** [comment]
> Closed in #2299 

@ColdHeat can you push an updated version of the image to Docker hub?
